core/src/main/java/org/apache/carbondata/core/datamap/DataMapUtil.java

```diff
@@ -91,43 +94,129 @@ public static DataMapJob getDataMapJob(Configuration configuration) throws IOExc
    * @param carbonTable
    * @throws IOException
    */
-  public static void executeDataMapJobForClearingDataMaps(CarbonTable carbonTable)
-      throws IOException {
-    String dataMapJobClassName = "org.apache.carbondata.spark.rdd.SparkDataMapJob";
+  private static void executeDataMapJobForClearingSegmentDataMaps(DataMapJob dataMapJob,
+      CarbonTable carbonTable) throws IOException {
+    SegmentStatusManager.ValidAndInvalidSegmentsInfo validAndInvalidSegmentsInfo =
+            getValidAndInvalidSegments(carbonTable, FileFactory.getConfiguration());
+    List<String> invalidSegment = new ArrayList<>();
+    for (Segment segment : validAndInvalidSegmentsInfo.getInvalidSegments()) {
+      invalidSegment.add(segment.getSegmentNo());
+    }
+    DistributableDataMapFormat dataMapFormat = new DistributableDataMapFormat(carbonTable, null,
```

Comment:
Instead of passing null for not required field can we expose another constructor to pass only the required non-null values and refactor the constructors within the class. Also move the null fields as the last arguments in the existing constructor

Reply:
done